examples/tree_1d_dgsem/elixir_shallowwater_multilayer_dam_break_ec.jl

```diff
@@ -0,0 +1,92 @@
+
+using OrdinaryDiffEq
+using Trixi
+using TrixiShallowWater
+
+###############################################################################
+# Semidiscretization of the multilayer shallow water equations for a dam break
+# test with a discontinuous bottom topography function to test entropy conservation
+
+equations = ShallowWaterMultiLayerEquations1D(gravity_constant = 9.81,
+                                              rhos = (0.85, 0.9, 1.0))
+
+# Initial condition of a dam break with a discontinuous water heights and bottom topography.
+# Works as intended for TreeMesh1D with `initial_refinement_level=5`. If the mesh
+# refinement level is changed the initial condition below may need changed as well to
+# ensure that the discontinuities lie on an element interface.
+function initial_condition_dam_break(x, t, equations::ShallowWaterMultiLayerEquations1D)
+    v = [0.0, 0.0, 0.0]
+
+    # Set the discontinuity
+    if x[1] <= 10.0
+        H = [6.0, 4.0, 2.0]
+        b = 0.0
+    else
+        H = [5.5, 3.5, 1.5]
+        b = 0.5
+    end
+
+    return prim2cons(SVector(H..., v..., b), equations)
+end
+
+initial_condition = initial_condition_dam_break
+
+###############################################################################
+# Get the DG approximation space
+
+volume_flux = (flux_ersing_etal, flux_nonconservative_ersing_etal)
+solver = DGSEM(polydeg = 3,
+               surface_flux = (flux_ersing_etal, flux_nonconservative_ersing_etal),
+               volume_integral = VolumeIntegralFluxDifferencing(volume_flux))
+
+###############################################################################
+# Get the TreeMesh and setup a non-periodic mesh
+
+coordinates_min = 0.0
+coordinates_max = 20.0
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 5,
+                n_cells_max = 10000,
+                periodicity = false)
+
+boundary_condition = boundary_condition_slip_wall
```

Comment:
The reason for the bounded domain is that I tried to cover as many features as possible to reduce the total amount of elixirs. I didn't see any problems with entropy conservation in this test, but I could just switch to periodic domains for this test.

Reply:
From a theory perspective, it is surprising that we get entropy conservation on the bounded domain. I wonder if it is because your EC test initializes zero velocities in all layers. Can you try a "harder" EC test and introduce jumps in the velocities as well? I suspect that this would trigger the wall BC and we would need to switch to a periodic test.

Reply:
Even if I initialize it with a velocity jump and run it for longer time `t=10` to ensure the wall boundary condition is actually used it remains EC.

Reply:
Okay, cool. Thanks for checking.